fairing/utils.py

```diff
@@ -33,3 +35,51 @@ def crc(file_name):
 
 def random_tag():
     return str(uuid.uuid4()).split('-')[0]
+
+
+def guess_preprocessor(entry_point, *args, **kwargs):
+    #Trying to avoid any clyclic dependencies by importing locally within the method
+    from fairing.functions.function_shim import get_execution_obj_type, ObjectType
+    from fairing.preprocessors.function import FunctionPreProcessor
+    
+    if get_execution_obj_type(entry_point) != ObjectType.NOT_SUPPORTED:
+        return FunctionPreProcessor(entry_point, *args, **kwargs)
+    else:
+        raise NotImplementedError(
+            "obj param should be a function or a class, got {}".format(type(entry_point)))
+
+
+def guess_docker_registry():
+    import fairing
+    try:
+        gcp_project = fairing.cloud.gcp.guess_project_name()
+        registry = 'gcr.io/{}'.format(gcp_project)
+        return registry
+    except Exception as e:
+        print(e)
+        return None
+
+
+def is_docker_daemon_exists():
```

Comment:
the utils package shouldn't have any dependencies on packages outside the stdlib

Reply:
otherwise you end up with a dependency on heavy clients like docker when you just need to generate a random UUID

Reply:
Ack. Moved util functions out of the common utils module.